### DISCUSSION TITLE
Edit multiple cells at once in $EDITOR

### DISCUSSION THREAD

**Discussion by Delapouite** [saulpw/visidata#1596] [Q&A]
Hello

Visidata lets the user edit the current cell in `$EDITOR` with `Ctrl+O` as documented on this page: https://www.visidata.org/docs/edit/ which is really handy, thanks.

Is there a way to do it for multiple cells at once?
Let's me illustrate by piggy-backing on the example data used in this tutorial : https://jsvine.github.io/intro-to-visidata/basics/opening-files/

Here's what the top of the sheet looks like : 
![image](https://user-images.githubusercontent.com/39315/201522340-4be33d4a-4c4b-4ecb-b52f-3c980a5bbb98.png)

Now I'm interested about editing the first row. I could press to `e` change the content of the first column, then move to the next column with `Shit+arrow keys` as described in https://github.com/saulpw/visidata/discussions/1321

It works well when the content of the cells is small. But can be cumbersome for big cells (even when expanded with `v`).

So if I press `Enter` on the first row of the sheet, I have a good overview of it : 
![image](https://user-images.githubusercontent.com/39315/201522433-9b0e0901-a890-4769-87e1-1f997a28f3bf.png)

This is the stage where I would find useful to have a command "Open this sheet in $EDITOR", which could result in something like that:
![image](https://user-images.githubusercontent.com/39315/201522849-e0b74bf4-633b-44f9-8b14-91f26586019a.png)

In the screenshot above, an external text-editor is opened (Kakoune here, but not relevant), each "column" is prefixed with a `#` and must be left untouched and each cell content is available just below.

When editing is finished, saving this temporary document would then applied back the changes to the original sheet. If ever by mistake a mismatch has been introduced, meaning a "column" has been removed or reordered by mistake for example, the changes would not be applied and en error message would be displayed.

Do you think such a feature could be possible?

Thanks for your ideas.

**Comment by frosencrantz** [comment]
Are you asking for a way to edit a row in your $EDITOR   I think such a feature would be useful if the text file being edited was in a standard format (think JSON or YAML)  Also there would be issues around rows with structured data like JSON

**Reply by frosencrantz** [reply]
@Delapouite would you be willing to code this idea into a VisiData plugin?   I think this would be a useful plugin.

I already have some feature ideas like:
- the ability to edit only the selected rows.
- the ability to head the values in a column
- how to handle new or missing columns in the output from your edits.

**Reply by Delapouite** [reply]
Sure, I'll try to draft a minimum case to get the ball rolling.

**Reply by saulpw** [reply]
Hey @Delapouite, this piqued my interest so I threw something together this evening. You might try the [`rec` format](https://www.gnu.org/software/recutils/), for example.  It only handles modifications currently; feel free to try to get it to do something sensible with added/deleted rows/columns. 

https://github.com/saulpw/visidata/commit/4ad4b92a48e87b3f4f9a5909386754e2407ed839
